src/main/java/org/craftercms/engine/store/s3/S3ContentStoreAdapter.java

```diff
@@ -98,16 +104,15 @@ public Context createContext(final String id, final String rootFolderPath, final
                                  final boolean ignoreHiddenFiles, Map<String, String> configurationVariables)
         throws RootFolderNotFoundException, StoreException, AuthenticationException {
 
-        AmazonS3URI uri = new AmazonS3URI(StringUtils.removeEnd(rootFolderPath, DELIMITER));
-
-        ListObjectsV2Request request = new ListObjectsV2Request()
-                                            .withBucketName(uri.getBucket())
-                                            .withPrefix(uri.getKey())
-                                            .withDelimiter(DELIMITER);
-        ListObjectsV2Result result = client.listObjectsV2(request);
+        S3Uri uri = client.utilities().parseUri(URI.create(StringUtils.removeEnd(rootFolderPath, DELIMITER)));
+        ListObjectsV2Request request = ListObjectsV2Request.builder()
+                .bucket(uri.bucket().orElseThrow())
```

Comment:
Should we throw a `NullPointerException` that indicates that the bucket is missing? I like more descriptive exceptions in general (I think this will only produce a `NullPointerException` with no message)

Reply:
I updated to use a new `S3BucketNotFoundException` and created a test case.